osbuild/objectstore.py

```diff
@@ -52,6 +53,35 @@ def umount(target, lazy=True):
     subprocess.run(["umount"] + args + [target], check=True)
 
 
+def remove_tree(path):
+    def fixperms(p):
+        subprocess.run(["chattr", "-i", p],
+                       check=False)
+        os.chmod(p, 0o700)
+
+    def unlink(p):
+        try:
+            os.unlink(p)
+        except IsADirectoryError:
+            remove_tree(p)
+        except FileNotFoundError:
+            pass
+
+    def on_error(_fn, p, exc_info):
+        e = exc_info[0]
+        if issubclass(e, FileNotFoundError):
+            pass
+        elif issubclass(e, PermissionError):
+            if p != path:
+                fixperms(os.path.dirname(p))
+            fixperms(p)
+            unlink(p)
+        else:
+            raise e
+
+    shutil.rmtree(path, onerror=on_error)
```

Comment:
This is modeled after what tmpfile does, right? Something like this:

https://github.com/python/cpython/blob/3.8/Lib/tempfile.py#L787

Reply:
Not sure I understand what you mean here ;)

Reply:
Ah, just wondered what the old code did and where this comes from. Nothing to act on.